packages/web-app-files/src/views/SharedWithMe.vue

```diff
@@ -110,50 +181,120 @@ export default {
+        return parseInt(this.$route.query['view-mode'])
+      }
+      return shareStatus.accepted
+    },
+
+    // pending shares
+    pendingSelected: {
       get() {
-        return this.selectedFiles
+        return this.selectedFiles.filter(r => r.status === shareStatus.pending)
       },
       set(resources) {
-        this.SET_FILE_SELECTION(resources)
+        // this will (intentionally) reset the file selection to pending shares only.
+        this.SET_FILE_SELECTION(resources.filter(r => r.status === shareStatus.pending))
       }
     },
+    pendingTitle() {
+      return this.$gettext('Pending shares')
+    },
+    pendingHasMore() {
+      return this.pendingCount > 3
+    },
+    pendingToggleMoreLabel() {
+      return this.showMorePending ? this.$gettext('Show less') : this.$gettext('Show more')
+    },
+    hasPending() {
+      return this.pendingCount > 0
+    },
+    pendingCount() {
+      return this.pending.length
+    },
+    pending() {
+      return this.activeFiles.filter(file => file.status === shareStatus.pending)
+    },
 
-    isEmpty() {
-      return this.activeFiles.length < 1
+    // accepted or declined shares
+    sharesSelected: {
+      get() {
+        return this.selectedFiles.filter(r => r.status === this.viewMode)
+      },
+      set(resources) {
+        // this will (intentionally) reset the file selection to shares for the current view mode only.
+        this.SET_FILE_SELECTION(resources.filter(r => r.status === this.viewMode))
+      }
+    },
+    sharesTitle() {
+      return this.viewMode === shareStatus.declined
+        ? this.$gettext('Declined shares')
+        : this.$gettext('Accepted shares')
+    },
+    sharesToggleLabel() {
+      return this.viewMode === shareStatus.declined
+        ? this.$gettext('Show accepted shares')
+        : this.$gettext('Show declined shares')
+    },
+    sharesEmptyMessage() {
+      return this.viewMode === shareStatus.declined
+        ? this.$gettext("You don't have any previously declined shares.")
+        : this.$gettext("You are not collaborating on other people's resources.")
+    },
+    hasShares() {
+      return this.sharesCount > 0
+    },
+    sharesCount() {
+      return this.shares.length
+    },
+    sharesCountFiles() {
+      return this.shares.filter(s => s.type !== 'folder').length
+    },
+    sharesCountFolders() {
+      return this.shares.filter(s => s.type === 'folder').length
+    },
+    shares() {
+      return this.activeFiles.filter(file => file.status === this.viewMode)
+    },
+    sharesOtherViewMode() {
+      return this.viewMode === shareStatus.accepted ? shareStatus.declined : shareStatus.accepted
+    },
+    sharesToggleRouterLink() {
+      return {
+        name: this.$route.name,
+        params: {
+          ...this.$route.params
+        },
+        query: {
+          ...this.$route.query,
+          'view-mode': this.sharesOtherViewMode
+        }
+      }
     },
 
+    // misc
```

Comment:
Dev leftover?

Reply:
nope, I tried to meta-group the computed props into props related to a) pending shares, b) accepted and declined shares and c) misc. ;-)